burnman/mineral.py

```diff
@@ -64,6 +65,15 @@ class which derives from the equation_of_state base class.
             raise Exception("Please derive your method from object (see python old style classes)")
 
         if self.method is not None and type(new_method) is not type(self.method):
+            # Reset everything but params
+            for key, elem in self.__dict__.items():
+                if key != 'params':
+                    elem = float('nan')
+
```

Comment:
This seems pretty dangerous.  Can we guarantee that there are no other members of the mineral that will be broken by this?  I'd rather not have it, I think.